core/src/main/scala/org/apache/spark/storage/BlockReplicationPolicy.scala

```diff
@@ -85,11 +65,9 @@ object BlockReplicationUtils {
    *         randomly shuffle elems
    */
   def getRandomSample[T](elems: Seq[T], m: Int, r: Random): List[T] = {
-    if (elems.size > m) {
-      getSampleIds(elems.size, m, r).map(elems(_))
-    } else {
-      r.shuffle(elems).toList
-    }
+    // This takes linear space, but is stable wrt m. That is for a fixed
```

Comment:
Hm, but I presume the purpose of the specialized method is to handle the case where elems is large. This isn't just test code so this might have a significant impact. Can the existing approach not be modified to be stable, or tests otherwise relaxed?

Reply:
Yes, the purpose was indeed that.

Do you think the contract tested in `BlockManagerReplicationBehavior` makes sense? Personally, I fail to see why this contract is important, but if there is a need I don't think we can do better than O(elems.size). Otherwise, the solution is to just remove the test.

Reply:
I see now. I suspect you should just remove the test instead. @shubhamchopra what do you think? I believe you created most of this code.

BTW if performance really mattered here we could make this sampling method a little bit more efficient (avoid foldLeft, -1 at the end, etc) but I don't think it matters

Reply:
@superbobry what do you think about just removing the test?

Reply:
Sounds good to me. Will push an update.